posthog/models/event/events_query.py

```diff
@@ -154,33 +150,40 @@ def run_events_query(
                 new_result["elements"] = ElementSerializer(
                     chain_to_elements(new_result["elements_chain"]), many=True
                 ).data
-            new_result["person"] = {
-                "id": new_result["person_id"],
-                "created_at": new_result["person.created_at"],
-                "properties": {
-                    "name": new_result["person.properties.name"],
-                    "email": new_result["person.properties.email"],
-                },
-                "distinct_ids": [new_result["distinct_id"]],
-            }
-            del new_result["person_id"]
-            del new_result["person.created_at"]
-            del new_result["person.properties.name"]
-            del new_result["person.properties.email"]
             query_result.results[index][star_idx] = new_result
 
-    # Convert person field from tuple to dict in each result
-    if "person" in select_input_raw:
+    if "person" in select_input_raw and len(query_result.results) > 0:
+        # Make a query into postgres to fetch person
         person_idx = select_input_raw.index("person")
-        for index, result in enumerate(query_result.results):
-            person_tuple: Tuple = result[person_idx]
-            query_result.results[index] = list(result)
-            query_result.results[index][person_idx] = {
-                "id": person_tuple[1],
-                "created_at": person_tuple[2],
-                "properties": {"name": person_tuple[3], "email": person_tuple[4]},
-                "distinct_ids": [person_tuple[0]],
-            }
+        distinct_ids = set(event[person_idx] for event in query_result.results)
+        persons = get_persons_by_distinct_ids(team.pk, list(distinct_ids))
+        persons = persons.prefetch_related(Prefetch("persondistinctid_set", to_attr="distinct_ids_cache"))
+        distinct_to_person: Dict[str, Person] = {}
+        for person in persons:
+            if person:
+                for distinct_id in person.distinct_ids:
+                    distinct_to_person[distinct_id] = person
+
+        # Loop over all columns in case there is more than one "person" column
+        for column_index, column in enumerate(select_input_raw):
+            if column != "person":
+                continue
+            for index, result in enumerate(query_result.results):
```

Comment:
I'm not super clear on what all is going on in these things yet 🙈 

This is an in-memory loop not an N+1 query?

Reply:
These are all in memory loops.

To get it to work, just select more than "last 24h"

Reply:
> To get it to work, just select more than "last 24h"

🙈 🤣 

Then I get

![Screenshot 2023-03-09 at 10 28 49](https://user-images.githubusercontent.com/984817/223997009-4c2b75f7-d2ae-4a5e-a42a-634a1306b1c8.png)

Reply:
`return !!personRecord.distinct_ids.length ? (`

when

```
{
    "uuid": "01869ed9-85a3-0000-f0dc-c6dbf095aef6",
    "created_at": "2023-03-01T20:24:59.852Z",
    "properties": {
        "name": null,
        "email": "paul@posthog.com"
    },
    "distinct_id": "a0JNkovhz54zqTBN2pBKf8jqA488qudRnugUzVshN2E"
}
```

Reply:
Dumping in 

```
else if (key === 'person' && isEventsQuery(query.source)) {
        const personRecord = value as PersonType
        let distinctId = undefined
        if (personRecord.distinct_ids && personRecord.distinct_ids.length > 0) {
            distinctId = personRecord.distinct_ids[0]
        } else if (personRecord.distinct_id) {
            distinctId = personRecord.id
        }
        return distinctId ? (
            <Link to={urls.person(distinctId)}>
                <PersonHeader noLink withIcon person={personRecord} />
            </Link>
        ) : (
            <PersonHeader noLink withIcon person={value} />
        )
    }
```

and the table displays (although the type system is unhappy)